File: cpp/ecbenchmark/src/function/Rosenbrock.cpp
#include "ecbenchmark/function/Rosenbrock.h"


/**
 * 
 * -2.048 - 2.048
 * 
 */
namespace ecb{
    scalar Rosenbrock::f(const std::vector<scalar>& x){
        scalar result = 0.0;
        for (size_t i = 0; i < x.size() - 1; ++i) {
            result += 100 * (x[i] * x[i] - x[i + 1]) * (x[i] * x[i] - x[i + 1]) + (x[i] - 1) * (x[i] - 1);
        }
        return result;
    }
}

File: java/ecbenchmark/src/ecbenchmark/function/Rosenbrock.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.function;

import ecbenchmark.Function;

/**
 *
 * @author jcrada
 */
public class Rosenbrock implements Function {

    public Rosenbrock() {
    }

    public double f(final double[] x) {
        double result = 0.0;
        for (int i = 0; i < x.length - 1; ++i) {
            result += 100 * (x[i] * x[i] - x[i + 1]) * (x[i] * x[i] - x[i + 1])
                    + (x[i] - 1) * (x[i] - 1);
        }
        return result;
    }
}
